Userland/Libraries/LibWeb/HTML/FormControlInfrastructure.cpp

```diff
@@ -150,9 +150,7 @@ WebIDL::ExceptionOr<Optional<HashMapWithVectorOfFormDataEntryValue>> construct_e
         // FIXME:    2. Create an entry with name and charset, and append it to entry list.
         // 10. Otherwise, create an entry with name and the value of the field element, and append it to entry list.
         else {
-            auto* input_element = dynamic_cast<HTML::HTMLInputElement*>(control.ptr());
-            VERIFY(input_element);
-            TRY_OR_THROW_OOM(vm, form_data_entries.try_append(input_element->value()));
+            TRY_OR_THROW_OOM(vm, form_data_entries.try_append(control->text_content()));
```

Comment:
This doesn't seem correct. `value` in the spec specifically links to https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-fe-value, which says:

> A control's [value](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-fe-value) is its internal state. As such, it might not match the user's current input.
> 
> For instance, if a user enters the word "three" into [a numeric field](https://html.spec.whatwg.org/multipage/input.html#number-state-(type=number)) that expects digits, the user's input would be the string "three" but the control's [value](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-fe-value) would remain unchanged. Or, if a user enters the email address `"  awesome@example.com"` (with leading whitespace) into [an email field](https://html.spec.whatwg.org/multipage/input.html#email-state-(type=email)), the user's input would be the string `"  awesome@example.com"` but the browser's UI for email fields might translate that into a [value](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-fe-value) of `"awesome@example.com"` (without the leading whitespace).

Reply:
Casting to a HTMLElement instead and retrieving the value attribute from there instead.